docs/sphinx/examples/cpp/algorithms/amplitude_estimation.cpp

```diff
@@ -93,58 +93,58 @@ struct run_circuit {
   auto operator()(const int n_qubits, const int n_itrs,
                   const double bmax) __qpu__ {
 
-    cudaq::qreg q(n_qubits + 1); // last is auxiliary
+    cudaq::qreg q(n_qubits + 1); /// last is auxiliary
     auto &last_qubit = q.back();
 
-    // State preparation
+    /// State preparation
     statePrep_A{}(q, bmax);
 
-    // Amplification Q^m_k as per evaluation schedule {m_0,m_1,..,m_k,..}
+    /// Amplification Q^m_k as per evaluation schedule {m_0,m_1,..,m_k,..}
     for (int i = 0; i < n_itrs; ++i) {
 
       z(last_qubit);
       cudaq::adjoint(statePrep_A{}, q, bmax);
       S_0{}(q);
       statePrep_A{}(q, bmax);
     }
-    // Measure the last auxiliary qubit
+    /// Measure the last auxiliary qubit
     mz(last_qubit);
   }
 };
 
 int main() {
 
-  const int n = 10; // number of qubits (not including the auxiliary qubit)
-  const double bmax = M_PI * 0.25; // upper bound of the integral
+  const int n = 10; /// number of qubits (not including the auxiliary qubit)
```

Comment:
I think this might get strange when Doxygen does its thing. IIRC, this comment will associate with the next line, the declaration of `bmax`. The comment on `bmax`'s line will associate (perhaps) with `schedule`, and so on.

Reply:
I agree this is strange to put it this way. This is way I was spellchecking normal comments - the examples are included in the documentation and as such should be checked. Making the comments there triple slash comments doesn't cause any issues since no doxygen comments are generated for them, but it is odd.

Reply:
This raises a good tangential point. We really should get the doxygen working. (I seem to recall it was at one point.) Looking at the actual output seems like a good idea while we're iterating on this.

Reply:
The complete doxygen documentation is generated [as part of CI](https://github.com/NVIDIA/cuda-quantum/blob/5c75d2437c5321157416b3a6a1679d2b39866371/.github/workflows/build_packages.yml#L136). That pipeline will fail if there are any warnings generated by doxygen (see the [doxygen configuration](https://github.com/NVIDIA/cuda-quantum/blob/5c75d2437c5321157416b3a6a1679d2b39866371/docs/Doxyfile.in#L885)). However, a couple of warnings are turned off, specifically the warnings for undocumented item or parameters. It would be good to turn these on eventually. 

While the complete doxygen documentation is always generated, not everything from it is included in the final docs page built using sphinx. I realized that while the html for the website is dumped by the build, the docs generated by doxygen are only downloadable from the artifacts when the build fails. I can change that, but maybe with another PR.

Reply:
It occurs to me that the generated xml for including the doxygen comments in sphinx is not very useful to look at. I can dump them in another format (e.g. html) as part of the build, but ultimately you would need to look at the sphinx documentation (which is dumped as artifact and pulls content from doxygen) to have an accurate preview of what the webpage will look like.

Reply:
Ok. Glad this is getting built.